### PR TITLE
#2502 add freePorts to WorkerClusterInfo

### PR DESCRIPTION
**Pull request by chia7712**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #2502)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

**Comment by chia7712** [comment]
retry configurator

**Comment by chia7712** [comment]
blocked by #2531

**Comment by chia7712** [comment]
retry it

**Comment by chia7712** [comment]
retry it

**Comment by chia7712** [comment]
unrelated error

File: docs/src/rest/rest-workers.rst [hunk 1]
```diff
@@ -43,6 +43,11 @@ The properties which can be set by user are shown below.
   #. brokerClusterName (**string**) — broker cluster used to host topics for this worker cluster
   #. clientPort (**int**) — worker client port
   #. jmxPort (**int**) — worker jmx port
+  #. freePorts (**Array(int)**) — thr ports you want to pre-bind for the connectors. If your connectors want
```
Comment (vitojeng) [inline docs/src/rest/rest-workers.rst]: The item of `freePorts` need reformat.
Please reference the item of `jarKeys`. :smile:

Comment (chia7712) [reply docs/src/rest/rest-workers.rst]: ok, I will address your comment. the #2502 is reopened .

**Pull request merged**